.github/workflows/ci.yml

```
@@ -100,6 +91,23 @@ jobs:
           # Make sure the help options print.
           python -m PyInstaller -h
 
+      - name: Install test dependencies (tools)
+        run: |
+          pip install --progress-bar=off -U -r tests/requirements-tools.txt
+
+      - name: Install test dependencies (libraries)
+        id: install_libraries
+        if: ${{ !endsWith(matrix.python-version, '-dev') }}
+        run: |
+          pip install --progress-bar=off -U -r tests/requirements-libraries.txt
```

```suggestion
          pip install --progress-bar=off -upgrade -r tests/requirements-libraries.txt
```
`pip install` has options: `--user`, `--upgrade`, `--upgrade-strategy`, and `--use-pep517` so let's not force the reader to guess.

---

Fair enough.